### ISSUE TITLE
[Idea/Improvement]  ARCH ~arm support

### ISSUE THREAD

**Issue by dguglielmi** [kodi-overlay/kodi-overlay#9]
As discussed earlier, we restrict ~arm because media-libs/libdisplay-info lake of support for KEYWORDS="~arm"

We depend on media-libs/libdisplay-info only if IUSE="gbm" and I saw that Libreelec have a build of libdisplay-info for ~arm (here on a raspberry pi 2)

```
readelf -a libdisplay-info.so.1
ELF Header:
  Magic:   7f 45 4c 46 01 01 01 00 00 00 00 00 00 00 00 00 
  Class:                             ELF32
  Data:                              2's complement, little endian
  Version:                           1 (current)
  OS/ABI:                            UNIX - System V
  ABI Version:                       0
  Type:                              DYN (Shared object file)
  Machine:                           ARM
  Version:                           0x1
  Entry point address:               0x0
  Start of program headers:          52 (bytes into file)
  Start of section headers:          205328 (bytes into file)
  Flags:                             0x5000400, Version5 EABI, hard-float ABI
  Size of this header:               52 (bytes)
  Size of program headers:           32 (bytes)
  Number of program headers:         7
  Size of section headers:           40 (bytes)
  Number of section headers:         27
  Section header string table index: 26
```

@parona-source : Is it just a missing KEYWORDS from ::gentoo or there is anything else that restrict us from using kodi on ~arm ?

**Comment by parona-source** [comment]
Its gentoo https://bugs.gentoo.org/930571, ill need to bother the arm  arch team to handle kodi + revdeps.

**Comment by parona-source** [comment]
https://bugs.gentoo.org/779184 Could you help with this if possible?

**Comment by parona-source** [comment]
Ok, I think I know the issue. Just bad assumptions from me.

**Comment by dguglielmi** [comment]
I don't have a 32 bits system right now, but I can install a Raspberry Pi 3 with Gentoo ARCH ~arm this week-end if needed.

I don't know if the following patch from Debian (related to time handling in Kodi) help or not.

- https://salsa.debian.org/multimedia-team/kodi-media-center/kodi/-/blob/debian/experimental/debian/patches/cdatetime-std-chrono/0001-Replace-Kodi-date-time-implementation-with-std-c.patch 

- The Upstream MR is not merged yet https://github.com/xbmc/xbmc/pull/18727

The patch is really big I don't know if we really want it downstream, but I can test.

It's not a fix for Y2038 problem for 32 bits system, but it maybe help to at least pass the tests. Do you think it can be helpful ?

**Comment by parona-source** [comment]
Ye, Im hesistant about something that large. Im currently thinking about just skipping the tests and documenting the issue

Thanks for pointing out that debian actually uses them.

**Comment by dguglielmi** [comment]
> Ye, Im hesistant about something that large. Im currently thinking about just skipping the tests and documenting the issue
> 
> Thanks for pointing out that debian actually uses them.

If Gentoo devs are OK with skipping these tests, I think it's the best way to deal with that for now.

**Comment by parona-source** [comment]
https://bugs.gentoo.org/930571#c1
https://gitweb.gentoo.org/repo/gentoo.git/commit/?id=f342cc586ca240c625e759034fe52a0de173c585

**Comment by dguglielmi** [comment]
Cool, we could try to add arm/arm64/riscv support for kodi addons now, what do you think ?

**Comment by parona-source** [comment]
You can test them here. I cant push them like that to ::gentoo of course due to the tree policies and needing the arch teams to sign off.

**Comment by dguglielmi** [comment]
Yep, I will have a look at this (maybe this week-end). Folks at libreelec have builds for arm/arm64. I think we can have a good part working here too
